cmd/imagec/main.go

```diff
@@ -195,11 +200,79 @@ func main() {
 		if err := ic.ListLayers(); err != nil {
 			log.Fatalf("Listing layers for image failed due to %s\n", err)
 		}
+	case Save:
+		options := imageCOptions.options
+
+		options.Outstream = os.Stdout
+
+		ap := archiveProxy(options.Host)
+		ic := imagec.NewImageC(options, streamformatter.NewJSONStreamFormatter())
+		if err := saveImage(ap, ic, options.Reference); err != nil {
+			log.Fatalf("Saving image %s failed due to %s\n", options.Reference.String(), err)
+		}
 	default:
 		log.Errorf("The operation '%s' is not valid\n", imageCOptions.operation)
 	}
 }
 
+func saveImage(ap proxy.VicArchiveProxy, ic *imagec.ImageC, ref reference.Named) error {
+	log.Debugf("Save image %s", ref.String())
+	err := ic.ListLayers()
+	if err != nil {
+		return err
+	}
+	layers, err := ic.LayersToDownload()
+	if err != nil {
+		return err
+	}
+
+	dest := imagec.DestinationDirectory(ic.Options)
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(layers))
+
+	for i := len(layers) - 1; i >= 0; i-- {
+		if layers[i].ID == imagec.ScratchLayerID {
+			continue
+		}
+		wg.Add(1)
+		go func(pid, cid string) {
+			defer wg.Done()
+			var err error
+			log.Debugf("parent id: %s, layer id: %s", pid, cid)
+
+			defer func() {
+				errChan <- err
+			}()
+
+			fileDir := path.Join(dest, cid)
+			err = os.MkdirAll(fileDir, 0755) /* #nosec */
```

Comment:
imagec pull will create folder for each layer file anyway.
I think it might be okay as long as we have same folder name for both imagec pull and image save.